controller/konnect/ops/ops_kongcertificate.go

```diff
@@ -0,0 +1,131 @@
+package ops
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"slices"
+
+	sdkkonnectcomp "github.com/Kong/sdk-konnect-go/models/components"
+	sdkkonnectops "github.com/Kong/sdk-konnect-go/models/operations"
+	sdkkonnecterrs "github.com/Kong/sdk-konnect-go/models/sdkerrors"
+	"github.com/samber/lo"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	ctrllog "sigs.k8s.io/controller-runtime/pkg/log"
+
+	configurationv1alpha1 "github.com/kong/kubernetes-configuration/api/configuration/v1alpha1"
+	"github.com/kong/kubernetes-configuration/pkg/metadata"
+)
+
+// createCertificate creates a KongCertificate in Konnect.
+// It sets the KonnectID and the Programmed condition in the KongCertificate status.
+func createCertificate(
+	ctx context.Context,
+	sdk CertificatesSDK,
+	cert *configurationv1alpha1.KongCertificate,
+) error {
+	cpID := cert.GetControlPlaneID()
+	if cpID == "" {
+		return fmt.Errorf("can't create %T %s without a Konnect ControlPlane ID", cert, client.ObjectKeyFromObject(cert))
+	}
+
+	resp, err := sdk.CreateCertificate(ctx,
+		cpID,
+		kongCertificateToCertificateInput(cert),
+	)
+
+	// TODO: handle already exists
+	// Can't adopt it as it will cause conflicts between the controller
+	// that created that entity and already manages it, hm
+	if errWrap := wrapErrIfKonnectOpFailed(err, CreateOp, cert); errWrap != nil {
+		SetKonnectEntityProgrammedConditionFalse(cert, "FailedToCreate", errWrap.Error())
+		return errWrap
+	}
+
+	cert.Status.Konnect.SetKonnectID(*resp.Certificate.ID)
+	SetKonnectEntityProgrammedCondition(cert)
+
+	return nil
+}
+
+// updateCertificate updates a KongCertificate in Konnect.
+// The KongCertificate must have a KonnectID set in its status.
+// It returns an error if the KongCertificate does not have a KonnectID.
+func updateCertificate(
+	ctx context.Context,
+	sdk CertificatesSDK,
+	cert *configurationv1alpha1.KongCertificate,
+) error {
+	cpID := cert.GetControlPlaneID()
+	if cpID == "" {
+		return fmt.Errorf("can't update %T without a ControlPlaneID", cert)
+	}
+
+	_, err := sdk.UpsertCertificate(ctx,
+		sdkkonnectops.UpsertCertificateRequest{
+			ControlPlaneID: cpID,
+			CertificateID:  cert.GetKonnectStatus().GetKonnectID(),
+			Certificate:    kongCertificateToCertificateInput(cert),
+		},
+	)
```

Comment:
To be consistent with all other implementations I suggest that we create the resource as a fall back when we receive a not found error, example for `KongService`: https://github.com/Kong/gateway-operator/blob/b449bc72d51a14c26bab4fe9bbbffefb13b93c2c/controller/konnect/ops/ops_kongservice.go#L75-L103

Reply:
Thank you